bin/citrea/tests/evm/mod.rs

```diff
@@ -224,6 +229,272 @@ async fn test_genesis_contract_call() -> Result<(), Box<dyn std::error::Error>>
+            let bytes = [
+                &[32], // balance length
+                acc_zkproof.balance.as_le_slice(),
+                &acc_zkproof.nonce.to_le_bytes(),
+                &code_hash_bytes,
+            ]
+            .concat();
+            Some(bytes)
+        } else {
+            // Account does not exist
+            None
+        };
+
+        let acc_proof: jmt::proof::SparseMerkleProof<sha2::Sha256> =
+            borsh::from_slice(&acc_zkproof.account_proof[0]).unwrap();
+
+        acc_proof
+            .verify(expected_root_hash, account_hash, proved_account)
+            .expect("Account proof must be valid");
+
+        for storage_proof in &acc_zkproof.storage_proof {
+            let storage_key = [
+                b"Evm/s/",
+                acc_zkproof.address.as_slice(),
+                &[32],
+                U256::from_le_slice(storage_proof.key.0.as_slice())
+                    .to_be_bytes::<32>()
+                    .as_slice(),
+            ]
+            .concat();
+            let key_hash = KeyHash::with::<sha2::Sha256>(storage_key.clone());
+
+            let proved_value = if storage_proof.proof[1] == Bytes::from("y") {
+                // Storage value exists and it's serialized form is:
+                let bytes = [&[32], storage_proof.value.to_be_bytes::<32>().as_slice()].concat();
+                Some(bytes)
+            } else {
+                // Storage value does not exist
+                None
+            };
+
+            let storage_proof: jmt::proof::SparseMerkleProof<sha2::Sha256> =
+                borsh::from_slice(&storage_proof.proof[0]).unwrap();
+
+            storage_proof
+                .verify(expected_root_hash, key_hash, proved_value)
+                .expect("Account storage proof must be valid");
+        }
+    }
+
+    let acc_zkproof_1 = seq_test_client
+        .eth_get_proof(
+            contract_address,
+            vec![contract_field],
+            Some(BlockNumberOrTag::Number(block_num)),
+        )
+        .await
+        .unwrap();
+
+    let acc_zkproof_2 = seq_test_client
+        .eth_get_proof(
+            contract_address,
+            vec![contract_field],
+            Some(BlockNumberOrTag::Number(block_num - 1)),
+        )
+        .await
+        .unwrap();
+
+    {
+        check_zk_proof(&acc_zkproof_1, contract_address);
+        for storage_proof in &acc_zkproof_1.storage_proof {
+            if U256::from_le_slice(storage_proof.key.0.as_slice()) == contract_field {
+                // A sanity check to verify we deal with the same value.
+                // This check is not actually required, it's for test purposes only
+                assert_eq!(storage_proof.value, storage_value);
+            }
+        }
+    }
+
+    {
+        // Assert historic proof is not the same as the first one queried
+        //  because storage root is different -> all proofs are different too.
+        assert_ne!(acc_zkproof_1, acc_zkproof_2);
```

Comment:
we must also assert the first proof queried is exactly the same as the newly queried one with rpc block num = block num at the time of sending the first one.